src/main/java/org/jenkinsci/lib/configprovider/AbstractConfigProviderImpl.java

```diff
@@ -110,6 +111,10 @@ protected XmlFile getConfigXml() {
         return new XmlFile(Jenkins.XSTREAM, new File(Jenkins.getActiveInstance().getRootDir(), this.getXmlFileName()));
     }
 
+    static {
+        Jenkins.XSTREAM.alias("org.jenkinsci.lib.configprovider.model.Config", CustomConfig.class);
```

Comment:
@ndeloof why do we need this alias now? - the class name/package did not change

Reply:
the issue here is that `Config` was not abstract and as such CasC don't know it has to search for concrete implementations. AFAICT `CustomConfig` is the derived class to best handle this case, but maybe a fresh new implementation would make sense ? In both cases for backward compatibility we need to tell xstream how to handle legacy `Config` instances.

Reply:
and the alias tells CasC to look for a concrete implementation? There are many different impls: `XmlConfig`, `JsonConfig`, `MavenToolchainsConfig`, `CustomConfig`, `GlobalMavenSettingsConfig`, `MavenSettingsConfig`, `GroovyScript` and others outside of this plugin. What would you expect from a "fresh new implementation"? what exact problem would it have to solve?

Reply:
ah now it makes sense - thanks for the clarification, I think `CustomConfig` is perfect for this